File: trip-kit/src/main/java/com/skedgo/android/tripkit/CarPark.java
package com.skedgo.android.tripkit;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

@Gson.TypeAdapters
@Value.Immutable
public abstract class CarPark {
  public abstract String identifier();
  public abstract String name();
  public abstract int totalSpaces();
  public abstract int availableSpaces();
  public abstract long lastUpdate();
}

File: TripKitData/src/main/java/com/skedgo/tripkit/data/database/locations/carparks/CarPark.kt
package com.skedgo.tripkit.data.database.locations.carparks

import com.google.gson.annotations.JsonAdapter

import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(GsonAdaptersCarPark::class)
interface CarPark {

    fun openingHours(): OpeningHours?
    fun pricingTables(): List<PricingTable>?
    fun operator(): ParkingOperator
    fun info(): String?
}